### PR TITLE
Add support for describing redshift clusters

### PR DESCRIPTION
**Pull request by edhollandAL**
@motobob @kkuzmin

### CODE REVIEW

File: src/erlcloud_redshift.erl [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+                                                  {error, term()}.
+describe_clusters(Options) when is_list(Options) ->
+    describe_clusters(Options, erlcloud_aws:default_config());
+describe_clusters(AwsConfig) ->
+    describe_clusters([], AwsConfig).
+
+-spec describe_clusters(list(), aws_config()) -> {ok, proplists:proplist()} |
+                                                 {ok, proplists:proplist(), list()} |
+                                                 {error, term()}.
+describe_clusters(Options, AwsConfig) ->
```
Comment (motobob) [inline src/erlcloud_redshift.erl:79]: please make the marker support correct/better.
http://docs.aws.amazon.com/redshift/latest/APIReference/API_DescribeClusters.html 
there should be following API:
describe_clusters().
describe_clusters(Cfg).
describe_clusters(Ids)
describe_clusters(Ids, Cfg)
describe_clusters(Ids, Options, Cfg).

describe_cluster_all(Options).
describe_cluster_all(Options, Cfg).

**Comment by edhollandAL** [comment]
```
0.12s$ wget https://s3.amazonaws.com/travis-otp-releases/binaries/$(lsb_release -is | tr "A-Z" "a-z")/$(lsb_release -rs)/$(uname -m)/erlang-18.1-nonroot.tar.bz2
--2016-01-06 19:20:37--  https://s3.amazonaws.com/travis-otp-releases/binaries/ubuntu/12.04/x86_64/erlang-18.1-nonroot.tar.bz2
Resolving s3.amazonaws.com (s3.amazonaws.com)... 54.231.11.83
Connecting to s3.amazonaws.com (s3.amazonaws.com)|54.231.11.83|:443... connected.
HTTP request sent, awaiting response... 403 Forbidden
2016-01-06 19:20:37 ERROR 403: Forbidden.
```

Nice test failure...

File: src/erlcloud_redshift.erl [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+       {vpc_security_groups, "VpcSecurityGroups/VpcSecurityGroup",
+        {optional_map, fun extract_vpc_security_groups/1}}
+      ],
+      Xml).
+
+describe_all(Fun, Options, Config, Acc) ->
+    describe_all(Fun, [], Options, Config, Acc).
+
+describe_all(Fun, Args, Options, Config, Acc) when is_list(Args) ->
+    case apply(Fun, Args ++ [Options] ++ [Config]) of
```
Comment (motobob) [inline src/erlcloud_redshift.erl:262]: just curious what is the point of this `Args` here as it's not really used? 
let it just be `Fun(Options, Cfg)` and if smb want to pass additional Args let is be via passing it like:

``` erlang
describe_all(fun(Opt, Cfg) -> some_other_fun(Arg1,Arg2, Opt, Cfg) end, Options, Cfg, Acc).
```

@edhollandAL obviously this is just a style observation.

Comment (edhollandAL) [inline src/erlcloud_redshift.erl:262]: i knicked this implementation from the route53 module where `Args` is used, i figured that leaving it in may be useful if someone adds more API methods later that do need it.

I hadn't considered the possibility the above suggestion when i originally implemented in the route53 module, happy to update both implementations to above style if prefered :)

File: src/erlcloud_redshift.erl [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+-spec(new/2 :: (string(), string()) -> aws_config()).
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{access_key_id=AccessKeyID,
+                secret_access_key=SecretAccessKey}.
+
+-spec(new/3 :: (string(), string(), string()) -> aws_config()).
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{access_key_id=AccessKeyID,
+                secret_access_key=SecretAccessKey,
+                route53_host=Host}.
```
Comment (kkuzmin) [inline src/erlcloud_redshift.erl]: redshift_host here.

File: src/erlcloud_redshift.erl (hunk 2 above)
Comment (motobob) [inline src/erlcloud_redshift.erl:262]: okay leave it like this and yes, it's time to commonize list_all in utils module

Comment (kkuzmin) [inline src/erlcloud_redshift.erl:262]: Is it possible to move `describe_all()` into for example `erlcloud_util.erl` in order to remove copy paste code?

Comment (edhollandAL) [inline src/erlcloud_redshift.erl:262]: Route53 version

```
describe_all(Fun, Args, Options, Config, Acc) when is_list(Args) ->
    case apply(Fun, Args ++ [Options] ++ [Config]) of
        {ok, Res} ->
            {ok, lists:foldl(fun erlang:'++'/2, [], [Res | Acc])};
        {ok, Res, {Name, Type}} ->
            Options1 = key_replace_or_add("name", Name, Options),
            Options2 = key_replace_or_add("type", Type, Options1),
            describe_all(Fun, Args, Options2, Config, [Res | Acc]);
        {ok, Res, Marker} ->
            Options1 = key_replace_or_add("marker", Marker, Options),
            describe_all(Fun, Args, Options1,
                         Config, [Res | Acc]);
        {error, Reason} ->
            {error, Reason}
    end;
describe_all(Fun, Args, Options, Config, Acc) ->
    describe_all(Fun, [Args], Options, Config, Acc).
```

Due to the differences in how markers are handled between different AWS APIs it's not a simple drop into `ercloud_util` jobbie, and much of the logic will still remain within each module so i'm not sure its worthwhile unless we can make it more generic at the same time

**Comment by kkuzmin** [comment]
+1

**Pull request merged**